tests/src/JIT/HardwareIntrinsics/X86/General/VectorArgs.cs

```diff
@@ -0,0 +1,114 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+
+// This test case is ported from S.N.Vector counterpart 
+// https://github.com/dotnet/coreclr/blob/master/tests/src/JIT/SIMD/VectorArgs.cs
+
+using System;
+using System.Runtime.Intrinsics;
+using System.Runtime.Intrinsics.X86;
+
+internal partial class IntelHardwareIntrinsicTest
+{
+    private const int Pass = 100;
+    private const int Fail = -1;
+
+    internal class VectorArg128
+    {
+        private Vector128<float> _rgb;
+
+        public VectorArg128(float r, float g, float b)
+        {
+            float[] temp = new float[4];
```

Comment:
Do the tests need to be a 1-to-1 mapping?

It seems like, in some cases, we could make these more intrinsic oriented (such as just using `SetVector128` here, instead of allocating an array)?

Reply:
Good point, will change.